docs/zkapps/o1js/foreign-fields.mdx

```diff
@@ -0,0 +1,111 @@
+---
+title: Foreign Field Arithmetic
+hide_title: true
+description: Foreign field arithmetic in o1js.
+keywords:
+  - foreign field
+  - non-native arithmetic
+  - wrong-field
+  - bigint
+  - o1js
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Foreign Field Arithmetic
+
+A foreign field is a [finite field](https://en.wikipedia.org/wiki/Modular_arithmetic) different from the native field of the proof system. o1js exposes operations like modular addition and multiplication that work in any finite field of size less than `2^259`.
+
+Foreign fields are useful for implementing cryptographic algorithms in provable code. For example, you use them for verification of Ethereum-compatible ECDSA signatures<!-- ([coming](https://github.com/o1-labs/o1js/pull/1007) [soon](https://github.com/o1-labs/o1js/pull/1291))-->.
```

Comment:
I've added comments to those pull requests to remind us to remove this when they land

Reply:
so we can remove the comments that you added @barriebyron?